src/core/components/files-mfs.js

```diff
@@ -3,38 +3,37 @@
 const promisify = require('promisify-es6')
 const mfs = require('ipfs-mfs/core')
 
-module.exports = self => {
-  const mfsSelf = Object.assign({}, self)
```

Comment:
@achingbrain I'm not entirely certain of all the consequences of doing this (given all the internal state that is kept by so many pieces. It doesn't look like a good practice or something that we want to debug later. If you know this is super safe, please do enlighten me as I would love to understand how.

On second note, what we really want is to avoid circular dependencies. js-ipfs-mfs should only take from IPFS what it needs and not the whole object.